### PR TITLE
Add support for L-inf distance in AKNN, custom scoring and painless scripting

### PR DESCRIPTION
**Pull request by elb3k**
Addition to this [pull request](https://github.com/opendistro-for-elasticsearch/k-NN/pull/310)

nmslib 2.0.11 added efficient indexing for `l1` and `linf`.
So I added support for l-inf spaceType, scoring and painless scripting.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java [hunk 1]
```diff
@@ -246,4 +246,42 @@ public static float l1distance(float[] queryVector, float[] inputVector) {
...
+     *           "query_vector": [1, 2, 3.4],
+     *           "field": "my_dense_vector"
+     *         }
+     *       }
+     *
+     * @param queryVector query vector
+     * @param docValues   script doc values
+     * @return L-inf score
+     */
+    public static float lInfDistance(List<Number> queryVector, KNNVectorScriptDocValues docValues) {
```
Comment (jmazanec15) [inline src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java]: Could you change this to lInfNorm? I think we are going to switch l1distance to l1Norm to follow that naming convention.

**Review by jmazanec15 — COMMENTED** [review]
Thanks for adding this!

**Review by jmazanec15 — APPROVED** [review]
LGTM!

**Review by VijayanB — APPROVED** [review]
LGTM

**Review by vamshin — APPROVED** [review]
LGTM! Thanks for adding new scoring functionality.

**Pull request merged**